saleor/graphql/payment/mutations.py

```diff
@@ -840,15 +854,17 @@ def perform_mutation(cls, root, info, **data):
             instance.save()
 
         transaction_event_data = data.get("transaction_event")
-        if instance.order_id and transaction_event_data:
-            transaction_event(
-                order=instance.order,
-                user=info.context.user,
-                app=info.context.app,
-                reference=transaction_event_data.get("reference", ""),
-                status=transaction_event_data.get("status", ""),
-                name=transaction_event_data.get("name", ""),
-            )
+        if transaction_event_data:
```

Comment:
```suggestion
        if transaction_event_data := data.get("transaction_event"):
```
and we can remove prev. line